### PR TITLE
feat: update blockchain API and redeploy contracts

### PR DESCRIPTION
**Pull request by Sokel**
-  new interface for simple-gatekeeper and bindings with deployed live and side gates
-  new interface OracleUSD and bindings with deployed contract
-  updating submodule needed for new topics
-  updating submodules with fixed contracts
-  rename some interfaces for consistent naming

### CODE REVIEW

File: blockchain/api.go [hunk 1]
```diff
@@ -875,3 +905,66 @@ func (api *BasicEventsAPI) processLog(log types.Log, eventTS uint64, out chan *E
...
 }
+
+type BasicSimpleGatekeeperApi struct {
+	client             *ethclient.Client
+	gatekeeperContract *marketAPI.Gatekeeper
+	gasPrice           int64
+	logParsePeriod     time.Duration
+}
+
+func NewBasicSimpleGatekeeperApi(client *ethclient.Client, address common.Address, gasPrice int64, logParsePeriod time.Duration) (SimpleGatekeeperAPI, error) {
```
Comment (sshaman1101) [inline blockchain/api.go]: I cannot get clearly - it is basic or it is simple? 
Also, `API` is an abbreviation and must be uppercased.

File: blockchain/api.go [hunk 2]
```diff
@@ -88,6 +90,12 @@ type TestTokenAPI interface {
 	GetTokens(ctx context.Context, key *ecdsa.PrivateKey) (*types.Transaction, error)
 }
 
+type SimpleGatekeeperAPI interface {
+	PayIn(ctx context.Context, key *ecdsa.PrivateKey, amount *big.Int) <-chan error
+
```
Comment (sshaman1101) [inline blockchain/api.go]: I'd personally prefer to have docstrings instead of blank lines.

File: blockchain/api.go (hunk 1 above)
Comment (Sokel) [reply blockchain/api.go]: Its Basic API for simple gatekeeper 🤦‍♂️

File: blockchain/api.go [hunk 3]
```diff
@@ -88,6 +91,17 @@ type TestTokenAPI interface {
...
 }
 
+type SimpleGatekeeperAPI interface {
+	PayIn(ctx context.Context, key *ecdsa.PrivateKey, value *big.Int) <-chan error
+	Payout(ctx context.Context, key *ecdsa.PrivateKey, to common.Address, value, txNumber *big.Int) <-chan error
+}
+
+type OracleUSDAPI interface {
+	SetCurrentPrice(ctx context.Context, key *ecdsa.PrivateKey, price *big.Int) (*types.Transaction, error)
+
```
Comment (sshaman1101) [inline blockchain/api.go]: I still prefer docstrings instead of blank lines.

File: blockchain/api.go [hunk 4]
```diff
@@ -134,7 +156,7 @@ func NewAPI(opts ...Option) (API, error) {
 		return nil, err
 	}
 
-	profileRegistry, err := NewProfileRegistry(clientSidechain, market.ProfileRegistryAddr(), defaults.gasPriceSidechain)
+	profileRegistry, err := NewBasicProfileRegistry(clientSidechain, market.ProfileRegistryAddr(), defaults.gasPriceSidechain)
```
Comment (sshaman1101) [inline blockchain/api.go]: What the reason to rename this?

File: blockchain/api.go [hunk 5]
```diff
@@ -88,6 +91,17 @@ type TestTokenAPI interface {
 	GetTokens(ctx context.Context, key *ecdsa.PrivateKey) (*types.Transaction, error)
 }
 
+type SimpleGatekeeperAPI interface {
+	PayIn(ctx context.Context, key *ecdsa.PrivateKey, value *big.Int) <-chan error
+	Payout(ctx context.Context, key *ecdsa.PrivateKey, to common.Address, value, txNumber *big.Int) <-chan error
+}
+
+type OracleUSDAPI interface {
```
Comment (sshaman1101) [inline blockchain/api.go]: You define an interface for the `Oracle` entity, but specify (via naming) that this interface implementation must manipulate only the USD currency. Is there any meaningful difference in a **interface**, if we want to receive the price in EUR?

File: blockchain/api.go (hunk 4 above)
Comment (Sokel) [reply blockchain/api.go]: Сonsistency&uniform

Comment (sshaman1101) [reply blockchain/api.go]: Out of scope.

**Pull request closed without merging**